File: algorithmique/fr-ioi/4.3_structures_donnees/peinture.cpp
#include <cstdio>

using namespace std;

int main() {
    int nbQte, qteCherchee, currQte;
    scanf("%d%d", &nbQte, &qteCherchee);
    bool isQuantite[100000000] = {false}; // Coup dur pour la RAM
    int quantites[100000];
    for (int i=0; i<nbQte; i++) {
        scanf("%d", &currQte);
        isQuantite[currQte] = true;
        quantites[i] = currQte;
    }
    for (int i =0; i<nbQte; i++) {
        if (isQuantite[qteCherchee-quantites[i]]) {
            printf("OUI\n");
            return 0;
        }
    }
    printf("NON\n");
}

File: algorithmique/fr-ioi/4.3_structures_donnees/peinture.py
import sys

def read():
    return sys.stdin.readline().strip()

def main():
    nbQte, qteCherchee = map(int, read().split())
    qtes = set(map(int, read().split()))
    for i in qtes :
        if qteCherchee-i in qtes :
            print("OUI")
            return
    print("NON")

main()